File: solutions/lua/grains/1/grains.lua
local grains = {}

function grains.square(n)
    local g = 1.0
    for i = 1, n - 1 do
        g = g + g
    end
    return g
end

function grains.total()
    return grains.square(64) * 2 - 1
end

return grains

File: solutions/julia/grains/1/grains.jl
"""Calculate the number of grains on square `square`."""
function on_square(square)
    if square < 1 || square > 64
        throw(DomainError(""))
    end
    sum::UInt64 = 1
    for i in range(1, square -1)
        sum = sum + sum
    end
    return sum
end

"""Calculate the total number of grains after square `square`."""
function total_after(square)
    total = on_square(square) * 2 - 1
end
